### PR TITLE
Refactor: redesign ColumnarTranspositionCipher

### PR DESCRIPTION
**Pull request by samuelfac**
Related to #5164

- [X] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Java/blob/master/CONTRIBUTING.md).
- [X] This pull request is all my own work -- I have not plagiarized it.
- [X] All filenames are in PascalCase.
- [x] All functions and variable names follow Java naming conventions.
- [x] All new algorithms have a URL in their comments that points to Wikipedia or other similar explanations.
- [X] All new code is formatted with `clang-format -i --style=file path/to/your/file.java`

### CODE REVIEW

**Comment by vil02** [comment]
Please focus on #5221 first.

File: src/main/java/com/thealgorithms/ciphers/ColumnarTranspositionCipher.java [hunk 1]
```diff
@@ -1,207 +1,123 @@
...
-    public static String encrpyter(String word, String keyword, String abecedarium) {
-        ColumnarTranspositionCipher.keyword = keyword;
-        ColumnarTranspositionCipher.abecedarium = Objects.requireNonNullElse(abecedarium, ABECEDARIUM);
-        table = tableBuilder(word);
-        Object[][] sortedTable = sortTable(table);
-        StringBuilder wordEncrypted = new StringBuilder();
-        for (int i = 0; i < sortedTable[0].length; i++) {
-            for (int j = 1; j < sortedTable.length; j++) {
-                wordEncrypted.append(sortedTable[j][i]);
+    public static String encrypt(String message, String key) {
```
Comment (vil02) [inline src/main/java/com/thealgorithms/ciphers/ColumnarTranspositionCipher.java:23]:
```suggestion
    public static String encrypt(final String message, final String key) {
```

File: src/main/java/com/thealgorithms/ciphers/ColumnarTranspositionCipher.java [hunk 2]
```diff
@@ -1,207 +1,123 @@
...
+        int numColumns = key.length();
+        int numRows = (int) Math.ceil((double) message.length() / numColumns);
```
Comment (vil02) [inline src/main/java/com/thealgorithms/ciphers/ColumnarTranspositionCipher.java:25]: This deserves a separate method, especially because it is used in `decrypt`.

File: src/main/java/com/thealgorithms/ciphers/ColumnarTranspositionCipher.java [hunk 3]
```diff
@@ -1,207 +1,123 @@
...
+     * @param message The encrypted message.
+     * @param key The key used for encryption.
+     * @return The decrypted message.
      */
-    public static String decrypter() {
-        StringBuilder wordDecrypted = new StringBuilder();
-        for (int i = 1; i < table.length; i++) {
-            for (Object item : table[i]) {
-                wordDecrypted.append(item);
+    public static String decrypt(String message, String key) {
```
Comment (vil02) [inline src/main/java/com/thealgorithms/ciphers/ColumnarTranspositionCipher.java:53]:
```suggestion
    public static String decrypt(final String message, final String key) {
```

File: src/test/java/com/thealgorithms/ciphers/ColumnarTranspositionCipherTest.java [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+        assertEquals(word, ColumnarTranspositionCipher.decrypt(encrypted, key));
+    }
+
+    private static Stream<Arguments> provideTestCases() {
+        return Stream.of(Arguments.of("asd215", "This is a test of the Columnar Transposition Cipher"), //
+            Arguments.of("test123456", "test"), //
+            Arguments.of("my long secret pass 1234567890", "short message"), //
+            Arguments.of("secret", "My secret message") //
+        );
+    }
```
Comment (vil02) [inline src/test/java/com/thealgorithms/ciphers/ColumnarTranspositionCipherTest.java:32]: These tests check only that `ColumnarTranspositionCipher.encrypt` is not an identity and that `encrypt` is an inverse of `decrypt`. I would highly suggest to:
- add the _expected_ into the test cases, i.e. each test case should consists of `message`, `key`, `encrypted`.
- remove the `encrypt` test (in its current form),
- add test cases with empty `word` and/or `key`,
- (as discussed earlier) remove the `//`

**Review by vil02 — CHANGES_REQUESTED** [review]

File: src/main/java/com/thealgorithms/ciphers/ColumnarTranspositionCipher.java (hunk 3 above)
Comment (SAIVARDHAN15) [reply src/main/java/com/thealgorithms/ciphers/ColumnarTranspositionCipher.java:53]: hey, I am currently working on #5223, I think the decrypt method does not need any parameters, because we have included global variables to store the respective..
